uaclient/apt.py

```diff
@@ -183,12 +187,14 @@ def run_apt_command(
 
 @lru_cache(maxsize=None)
 def run_apt_cache_policy_command(
+    package: Optional[str] = None,
     error_msg: Optional[str] = None,
     env: Optional[Dict[str, str]] = {},
 ) -> str:
-    return run_apt_command(
-        cmd=["apt-cache", "policy"], error_msg=error_msg, env=env
-    )
+    cmd = ["apt-cache", "policy"]
+    if package:
+        cmd.append(package)
+    return run_apt_command(cmd, error_msg=error_msg, env=env)
```

Comment:
`apt-cache policy` has drastically different output with/without a package argument. Should we model this as two separate functions in our code?

I could be convinced either way

Reply:
IMHO the idea of the function is just to wrap the call to policy, and mainly cache it so we don't need to run it more than once per execution.
I think we should keep this one function, for simplicity's sake: the return value is a string in both cases, and I don't think we gain much for adding the granularity here.

Reply:
I don't know, I tend to agree with @orndorffgrant here. Because the output is so distinct, I think it would be better for us to have a new function just to be explicit. Additionally, even though this is a minor thing, but I don't think caching will be that beneficial for the `apt-cache policy package` command

Reply:
oki can do